src/rgw/store/dbstore/common/dbstore.h

```diff
@@ -592,6 +589,22 @@ class DBOp {
       FOREIGN KEY (BucketName) \
       REFERENCES '{}' (BucketName) ON DELETE CASCADE ON UPDATE CASCADE \n);";
 
+    const std::string CreateObjectViewQ =
+      /* This query creats temporary view with entries from ObjectData table which have
+       * corresponding head object (i.e, with same ObjName, ObjInstance, ObjNS, ObjID)
+       * in the Object table.
+       *
+       * GC thread can use this view to delete stale entries from the ObjectData table which
+       * do not exist in this view.
+       *
+       * XXX: This view is throwing ForeignKey mismatch error, mostly may be because all the keys
+       * of objectdata table are not referenced here. So this view is not used atm.
+       */
+      "CREATE TEMP VIEW IF NOT EXISTS '{}' AS \
```

Comment:
this is getting rather database implementation specific (temp view, views in general);  it's elegant, but I am sort of unclear why it's really necessary; e.g., on a specific system (ok, sqlite3) is it measurably faster?

Reply:
(as we rise up the implementation-specific feature mountain, we eventually come to materialized views;  these vary from system to system, and could improve or hinder perf, depending on workload)

Reply:
okay thanks! will make a note. I am not sure if they are faster. This view is not used anywhere right now but have added it just as placeholder to use later (in case this query had to be repeated for multiple operations/instances).